parser.y

```diff
@@ -471,62 +482,58 @@ Term {
 }
 
 Import:
-"import" String "as" '$' IDENT ';' {
+ImportWhat ';' {
+  $$ = $1;
+} |
+ImportWhat Exp ';' {
+  if (!block_is_const($2)) {
+    FAIL(@$, "Module metadata must be constant");
+    $$ = gen_noop();
+    block_free($1);
+    block_free($2);
+  } else if (block_const_kind($2) != JV_KIND_OBJECT) {
+    FAIL(@$, "Module metadata must be an object");
```

Comment:
Ditto here: module and import metadata should be allowed to be things other than objects.